docs/contribution_policy.md

```diff
@@ -0,0 +1,111 @@
+in the EESSI software layer.
+
+## Requirements
+
+The following requirements must be taken into account when adding software to EESSI.
+
+---
+
+### a) Open source software { #open_source_software }
+
+Only **open source software** can be added to the EESSI repository.
+
+Make sure that you are aware of relevant software license, and that redistribution is allowed.
+
+For more information about a specific license, see the [SPDX license list](https://spdx.org/licenses/).
+
+!!! note
+
+    We intend to automatically verify that this requirement is met,
+    by requiring that the [SPDX license identifier](https://spdx.dev/ids/) is provided for all software.
+
+
+---
+
+### b) Built by the bot { #built_by_bot }
+
+All software included in the EESSI repository *must* be built autonomously by [our bot :robot:](bot.md),
+see also the [semi-automatic software installation procedure](software_layer/adding_software.md).
+
+
+---
+
+### c) Supported by EasyBuild { #supported_by_easybuild }
+
+Currently, we require that all software being added to EESSI is supported by the *EasyBuild release* being used
+to perform the installation.
+
+That is, the easyconfig files used for the installation *must be included in the EasyBuild release*.
+
+We do allow the use of [`--from-pr`](https://docs.easybuild.io/integration-with-github/#github_from_pr) and
+[`--include-easyblocks-from-pr`](https://docs.easybuild.io/integration-with-github/#github_include_easyblocks_from_pr)
+to pull in changes required to make the installation work correctly in the EESSI build environment,
+but only if that is strictly required.
+
+
+---
+
+### d) Compiler toolchain { #compiler_toolchain }
+
+A [compiler toolchain](https://docs.easybuild.io/terminology/#toolchains) that is still supported by the latest
+EasyBuild must be used for building the software.
+
+More information on supported toolchains in EasyBuild is available [here](https://docs.easybuild.io/policies/toolchains).
+
+
+---
+
+### e) CPU targets { #cpu_targets }
+
+The software *should* work on all [CPU targets supported by EESSI](software_layer/cpu_targets.md).
+
+Exceptions to this requirement are allowed if technical problems that can not be resolved with reasonable effort
+prevent the installation of the software for specific CPU targets.
+
+
+---
+
+### f) Versions & toolchains { #versions_toolchains }
+
+Recent software versions and toolchains *should* be preferred,
+although the installation of older software versions and toolchains is allowed if sufficiently motivated.
+
+
+---
+
+### g) Testing { #testing }
+
+We should be able to test the software installations via the [EESSI test suite](../test-suite).
+
+Ideally one or more tests are available that verify that the software is functionally correct,
+and performs well.
+
+It should be possible to run a minimal *smoke test*, for example using EasyBuild's `--sanity-check-only` feature.
```

Comment:
Making this a requirement for someone who is really just an end-user is a high bar. I agree for that marquee applications we should have such tests, but it's not realistic to expect that for all cases. In general, it's probably a much more pro-active effort on our part where we'd need to ensure we are reaching out to the actual developers (or very experienced users)

Reply:
This indeed necessarily needs to be a rather loose requirement, or we will be way too restrictive, and won't get any contributions because of that.

It's definitely unreasonable to require that a test must be available in the EESSI test suite, since adding something there is far from trivial (requires knowing Python, ReFrame, how to write a portable test, etc.).
We can probably provide some documentation with guidelines on how to do that eventually, but then it will still be quite an elaborate task.

We should phrase this such that it's clear that we prefer being able to test the software, in one way or another, without imposing a huge amount of effort on contributors, at least initially.
Maybe we can mention options like providing a test case, referring to an example or documentation of a basic run, etc.?